simd/unit_tests/include/TestSIMD_MathOps.hpp

```diff
@@ -73,40 +74,70 @@ void host_check_math_op_one_loader(UnaryOp unary_op, std::size_t n,
+  host_check_math_op_one_loader_unary<Abi, load_as_scalars>(op, n, arg, ulps);
+}
+
+template <class Abi, class Op, class T>
+inline void host_check_math_op_all_loaders_binary(Op op, std::size_t n,
+                                                  T const* arg1, T const* arg2,
+                                                  double ulps) {
+  host_check_math_op_one_loader_binary<Abi, load_element_aligned>(op, n, arg1,
+                                                                  arg2, ulps);
+  host_check_math_op_one_loader_binary<Abi, load_masked>(op, n, arg1, arg2,
+                                                         ulps);
+  host_check_math_op_one_loader_binary<Abi, load_as_scalars>(op, n, arg1, arg2,
+                                                             ulps);
 }
 
 template <typename Abi, typename DataType, size_t n>
 inline void host_check_all_math_ops(const DataType (&first_args)[n],
                                     const DataType (&second_args)[n]) {
-  host_check_math_op_all_loaders<Abi>(plus(), n, first_args, second_args);
-  host_check_math_op_all_loaders<Abi>(minus(), n, first_args, second_args);
-  host_check_math_op_all_loaders<Abi>(multiplies(), n, first_args, second_args);
-  host_check_math_op_all_loaders<Abi>(absolutes(), n, first_args);
-
-  host_check_math_op_all_loaders<Abi>(floors(), n, first_args);
-  host_check_math_op_all_loaders<Abi>(ceils(), n, first_args);
-  host_check_math_op_all_loaders<Abi>(rounds(), n, first_args);
-  host_check_math_op_all_loaders<Abi>(truncates(), n, first_args);
+  const double integral_ulps = 0;
+  host_check_math_op_all_loaders_binary<Abi>(plus(), n, first_args, second_args,
+                                             integral_ulps);
+  host_check_math_op_all_loaders_binary<Abi>(minus(), n, first_args,
+                                             second_args, integral_ulps);
+  host_check_math_op_all_loaders_binary<Abi>(multiplies(), n, first_args,
+                                             second_args, integral_ulps);
+  host_check_math_op_all_loaders_unary<Abi>(absolutes(), n, first_args,
+                                            integral_ulps);
+
+  host_check_math_op_all_loaders_unary<Abi>(floors(), n, first_args,
+                                            integral_ulps);
+  host_check_math_op_all_loaders_unary<Abi>(ceils(), n, first_args,
+                                            integral_ulps);
+  host_check_math_op_all_loaders_unary<Abi>(rounds(), n, first_args,
+                                            integral_ulps);
+  host_check_math_op_all_loaders_unary<Abi>(truncates(), n, first_args,
+                                            integral_ulps);
 
   // TODO: Place fallback implementations for all simd integer types
   if constexpr (std::is_floating_point_v<DataType>) {
-    host_check_math_op_all_loaders<Abi>(divides(), n, first_args, second_args);
-
-#if defined(__INTEL_COMPILER) && \
+    const double floating_ulps = 4;
```

Comment:
4ULPs is what `EXPECT_FLOAT_EQ` and `EXPECT_DOUBLE_EQ` use.
Why are we rolling our own floating-point comparison?

Reply:
Mostly so we can adjust the `ulps` when necessary and are explicit about what tolerances we accept.